src/inet/TCPEndPoint.cpp

```diff
@@ -1640,11 +1705,26 @@ void TCPEndPoint::TCPUserTimeoutHandler(Weave::System::Layer* aSystemLayer, void
 
 }
 
+#if INET_CONFIG_ENABLE_POLL_TCP_SEND_QUEUE
+void TCPEndPoint::ScheduleNextTCPUserTimeoutPoll()
+{
+    Weave::System::Layer& lSystemLayer = SystemLayer();
+
+    lSystemLayer.StartTimer(mTCPSendQueuePollPeriodMillis, TCPUserTimeoutHandler, this);
+}
+#endif // INET_CONFIG_ENABLE_POLL_TCP_SEND_QUEUE
+
 void TCPEndPoint::StartTCPUserTimeoutTimer()
 {
     Weave::System::Layer& lSystemLayer = SystemLayer();
 
+#if INET_CONFIG_ENABLE_POLL_TCP_SEND_QUEUE
+    mMaxTCPSendQueuePolls = mUserTimeoutMillis / mTCPSendQueuePollPeriodMillis;
+
+    lSystemLayer.StartTimer(mTCPSendQueuePollPeriodMillis, TCPUserTimeoutHandler, this);
```

Comment:
Yes, this could be replaced with the call to ScheduleNextTCPUserTimeoutPoll(). That way, the StartTimer(mTCPSendQueuePollPeriodMillis, ...) call can only be inside that function.